src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/DataElement.js

```diff
@@ -0,0 +1,228 @@
+// @flow
+/* eslint-disable no-underscore-dangle */
+import log from 'loglevel';
+import i18n from '@dhis2/d2-i18n';
+import { pipe, errorCreator } from 'capture-core-utils';
+
+import type {
+    CachedProgramTrackedEntityAttribute,
+    CachedTrackedEntityAttribute,
+    CachedOptionSet,
```

Comment:
I think we should define our own types for this particular context. There can be differences between what is in the cache and the api data we use here.